guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc

```diff
@@ -0,0 +1,237 @@
+database using Micronaut Data JDBC.
+
+include::{commondir}/common-requirements.adoc[]
+
+include::{commondir}/common-completesolution.adoc[]
+
+include::{commondir}/common-create-app.adoc[]
+
+
+=== Data Source configuration
+
+Add the following dependencies:
+
+:dependencies:
+
+
+implementation("io.micronaut:micronaut-http-client")
+implementation("io.micronaut:micronaut-runtime")
+implementation("io.micronaut.data:micronaut-data-hibernate-jpa")
+implementation("io.micronaut.flyway:micronaut-flyway")
+
+dependency:micronaut-data-processor[groupId=io.micronaut.data,scope=annotationProcessor,callout=1]
+dependency:micronaut-data-jdbc[groupId=io.micronaut.data,callout=2]
+dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=3]
+dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=4]
+
+:dependencies:
+
+<1> Add Micronaut Data annotation processor.
+<2> Configures  a compile scoped dependency on the `micronaut-data-jdbc`
+<3> Configures SQL DataSource instances using Hikari Connection Pool.
+<4> Add dependency to in-memory H2 Database.
+
+Define the data source in `src/main/resources/application.yml`.
+
+resource:application.yml[tag=datasource]
+
+NOTE: This way of defining the datasource properties means that we can externalize the configuration, for example for
+production environment, and also provide a default value for development. If the environment variables are not defined
+Micronaut will use the default values. +
+Also keep in mind that it is necessary to escape the `:` in the connection url using back ticks ```.
+
+=== DB Schema
+
+Now that Mybatis is setup we need a way to create the database schema. For that we will use
+https://micronaut-projects.github.io/micronaut-flyway/latest/guide/index.html[Micronaut integration with
+Flyway].
+
+
+Add the following snippet to include the necessary dependencies:
+
+dependency:micronaut-flyway[groupId=io.micronaut.flyway]
+
+Configure the database migrations directory for Flyway in `application.yml`.
+
+resource:application.yml[tag=flyway]
+
+Create the file `V1__schema.sql` with the database schema creation:
+
+resource:db/migration/V1__schema.sql[]
+
+During application startup Flyway will execute the sql file and create the schema needed for the application.
+
+=== Domain
+
+To use JPA annotations include the following `compileOnly` scoped dependency:
+
+dependency:jakarta.persistence-api[groupId=jakarta.persistence,version=@jakarta.persistence-apiVersion@,scope=compileOnly,callout=1]
+
+Create the domain entities:
+
+source:domain/Genre[]
+
+The previous domain has a `OneToMany` relationship with the domain `Book`.
+
+source:domain/Book[]
+
+
+=== Application Configuration
+
+Create an interface to encapsulate the application configuration settings:
+
+source:ApplicationConfiguration[]
+
+In Micronaut, like Spring Boot and Grails, you can create type safe configuration by creating classes that are annotated
```

Comment:
No need to mention spring and grails imo

Reply:
removed